source/include/VirtualInstrument.h

```diff
@@ -140,11 +145,10 @@ class VIClump : public FunctionClump
     VirtualInstrument*  TopVI()      {
         VIClump *caller = this;
         do {
-            while (caller->_waitingClumps)
```

Comment:
Was the old logic of using _waitingClumps incorrect? That you ran into while implementing callchain? Or, is it some optimization?